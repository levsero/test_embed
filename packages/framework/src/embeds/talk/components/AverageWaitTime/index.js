import PropTypes from 'prop-types'
import styled from 'styled-components'
import { FONT_SIZE } from 'src/constants/shared'
import { TEST_IDS } from 'src/constants/shared'

const AverageWaitTime = styled.p.attrs(() => ({ testId: TEST_IDS.TALK_AVG_WAIT_TIME }))`
  margin-bottom: ${15 / FONT_SIZE}rem;
  data-testid: ${(props) => props.testId};
`

AverageWaitTime.propTypes = {
  children: PropTypes.string,
}

export default AverageWaitTime
