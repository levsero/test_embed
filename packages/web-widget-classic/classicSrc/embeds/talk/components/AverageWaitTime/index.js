import { FONT_SIZE } from 'classicSrc/constants/shared'
import { TEST_IDS } from 'classicSrc/constants/shared'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const AverageWaitTime = styled.p.attrs(() => ({ testId: TEST_IDS.TALK_AVG_WAIT_TIME }))`
  margin-bottom: ${15 / FONT_SIZE}rem;
  data-testid: ${(props) => props.testId};
`

AverageWaitTime.propTypes = {
  children: PropTypes.string,
}

export default AverageWaitTime
