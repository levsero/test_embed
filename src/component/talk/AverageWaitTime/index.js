import React from 'react';
import { locals as styles } from './styles/index.scss';
import PropTypes from 'prop-types';

const AverageWaitTime = ({ message }) => {
  return <p className={styles.message}>{message}</p>;
};

AverageWaitTime.propTypes = {
  message: PropTypes.string
};

export default AverageWaitTime;
