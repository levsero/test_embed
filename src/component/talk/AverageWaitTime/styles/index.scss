@import '../../../../styles/_vars.scss';

.message {
  margin-bottom: #{15 / $font-size}rem;
}
