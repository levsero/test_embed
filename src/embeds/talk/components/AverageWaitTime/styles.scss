@import 'src/styles/_vars.scss';

.message {
  margin-bottom: #{15 / $font-size}rem;
}
