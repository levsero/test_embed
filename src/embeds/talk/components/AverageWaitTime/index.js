import React from 'react'
import PropTypes from 'prop-types'

import { locals as styles } from './styles.scss'

const AverageWaitTime = ({ message }) => {
  return <p className={styles.message}>{message}</p>
}

AverageWaitTime.propTypes = {
  message: PropTypes.string
}

export default AverageWaitTime
