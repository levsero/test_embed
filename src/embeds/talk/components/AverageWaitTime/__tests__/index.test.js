import { render } from '@testing-library/react'
import React from 'react'

import AverageWaitTime from '../'

test('renders the component', () => {
  const { container } = render(<AverageWaitTime>Average wait time: 1 minute</AverageWaitTime>)

  expect(container).toMatchSnapshot()
})
