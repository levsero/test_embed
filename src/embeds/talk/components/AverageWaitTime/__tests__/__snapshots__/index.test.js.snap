// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div>
  <p
    class="sc-bdVaJa ktbwkF"
  >
    Average wait time: 1 minute
  </p>
</div>
`;
